src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java

```diff
@@ -0,0 +1,129 @@
+package at.ac.fhcampuswien.fhmdb.models;
+
+import at.ac.fhcampuswien.fhmdb.filter.Genre;
+import com.j256.ormlite.field.DatabaseField;
+import com.j256.ormlite.table.DatabaseTable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@DatabaseTable(tableName = "watchlist")
+public class WatchlistEntity {
+    @DatabaseField(generatedId = true)
+    private long id;
+
+    @DatabaseField
+    private String apiId;
+
+    @DatabaseField
+    private String title;
+
+    @DatabaseField
+    private String description;
+
+    @DatabaseField
+    private String genres;
+
+    @DatabaseField
+    private int releaseYear;
+
+    @DatabaseField
+    private String imgUrl;
+
+    @DatabaseField
+    private int lengthInMinutes;
+
+    @DatabaseField
+    private String directors;
+
+    @DatabaseField
+    private String writers;
+
+    @DatabaseField
+    private String mainCast;
+
+    @DatabaseField
+    private double rating;
+
+    public WatchlistEntity() {
+        // ORMLite needs a no-arg constructor
+    }
+
+    public WatchlistEntity(Movie movie) {
+        this.apiId = movie.getId();
+        this.imgUrl = movie.getImgUrl();
+        this.title = movie.getTitle();
+        this.description = movie.getDescription();
+        this.genres = genreListToString(movie.getGenres());
+        this.releaseYear = movie.getReleaseYear();
+        this.directors = listToString(movie.getDirectors());
+        this.writers = listToString(movie.getWriters());
+        this.mainCast = listToString(movie.getMainCast());
```

Comment:
According to the Class diagram, these fields are not needed, or am I missing something?

Reply:
I am mapping the data to the Database fields. As Ive understood it you have to declare these database fields and then map the data to them.

fe.:
@DatabaseField
    private String directors;

    @DatabaseField
    private String writers;

    @DatabaseField
    private String mainCast;

    @DatabaseField
    private double rating;

Reply:
and we cannot save a list in the database. We save it as a String with the comma delimiter

Reply:
I understand, but in the Angabe it says "Da für das Speichern von Listen (zB directors, writers, mainCast) weitere Tabellen nötig wären, werden diese ausgenommen.", that's why I asked

Reply:
I dient want to have the same shitty situation as last time with the UI thing where we lost points so I have implemented it ^^.

Reply:
Fair enough 😁